### PR TITLE
add build static library script

### PR DESCRIPTION
**Pull request by nmreadelf**
add build static library

### CODE REVIEW

**Comment by nmreadelf** [comment]
static library lost some functions register, I need help

File: chdb/build.sh [hunk 1]
```diff
@@ -123,8 +123,68 @@ if [ "$(uname)" == "Linux" ]; then
 fi
 
 # save the command to a file for debug
+echo ${LIBCHDB_CMD} > libchdb_cmd.sh.origin
+LIBCHDB_CMD=$(gawk '{for(i=1;i<NF;i++){if ($i !~ /\.o$/){printf "%s ", $i}}}END{printf "\n"}' libchdb_cmd.sh.origin)
+# generate libchdb.a
+object_list=$(gawk '{for(i=0;i<NF;i++){if ($i ~ /\.o$/){printf "%s ", $i}}}END{printf "\n"}' libchdb_cmd.sh.origin)
+LIBCHDB_STATIC_CMD=$(echo "llvm-ar-15 rsc libchdb.a ${object_list}")
```
Comment (auxten) [inline chdb/build.sh:130]: We are using llvm 16

File: chdb/build.sh [hunk 2]
```diff
@@ -123,8 +123,68 @@ if [ "$(uname)" == "Linux" ]; then
...
+  rm -f "${temp_working_dir}/\${prefix}" || :
+  #ar -t ${f} > "/tmp/${prefix}.list"
+done
+llvm-ar-15 -csr chdb.a ${temp_working_dir}/*.o
+ls -lh chdb.a
+rm -rf ${temp_working_dir}
+EOF
+#------ generate script to create chdb.a   end--------
+
+LIBCHDB_CMD=$(echo ${LIBCHDB_CMD} | gawk '{for(i=1;i<NF;i++){if ($i !~ /\.a$/){printf "%s ", $i; if ($i ~ /_chdb.cpython-38-x86_64-linux-gnu.so/){printf "chdb.a "}}}}END{printf "\n"}')
```
Comment (auxten) [inline chdb/build.sh:181]: It's not always py38

File: chdb/build.sh [hunk 3]
```diff
@@ -123,8 +123,68 @@ if [ "$(uname)" == "Linux" ]; then
 fi
 
 # save the command to a file for debug
+echo ${LIBCHDB_CMD} > libchdb_cmd.sh.origin
```
Comment (auxten) [inline chdb/build.sh:126]: Maybe we can just pust static lib gen stuff into a separate `build_static.sh` fow now

**Comment by 0x00F6** [comment]
Hello,

What is missing to accept this pull request?

**Comment by nmreadelf** [comment]
> Hello,
> 
> What is missing to accept this pull request?

https://github.com/chdb-io/chdb/pull/121#issuecomment-1774343543

**Pull request closed without merging**